architecture/evm/error_normalizer.go

```diff
@@ -257,66 +301,78 @@ func ExtractJsonRpcError(r *http.Response, nr *common.NormalizedResponse, jr *co
+			strings.Contains(msg, "Unsupported method") ||
+			strings.Contains(msg, "not supported") ||
+			strings.Contains(msg, "method is not whitelisted") ||
+			strings.Contains(msg, "is not included in your current plan") {
+			return common.NewErrEndpointUnsupported(
 				common.NewErrJsonRpcExceptionInternal(
 					int(code),
-					common.JsonRpcErrorInvalidArgument,
+					common.JsonRpcErrorUnsupportedException,
 					err.Message,
 					nil,
 					details,
 				),
 			)
-		} else if r.StatusCode == 415 || code == -32004 || code == -32001 || code == common.JsonRpcErrorUnsupportedException {
-			return common.NewErrEndpointUnsupported(
+		}
+
+		//----------------------------------------------------------------
+		// "Invalid Argument / Params / Request" errors
+		//----------------------------------------------------------------
+
+		// Even though these errors have invalid argument or params, they are more about a lack of standard
+		// for certain args/params for certain methods, among different providers or clients. We will retry them.
+		//
+		// Examples:
+		if strings.Contains(msg, "tx of type") || // Should be retried toward upstreams that support this tx type
+			strings.Contains(msg, "invalid type: map, expected BlockNumber, 'latest', or 'earliest'") { // Envio not supporting { blockNumber: XXX, blockHash: YYY } for eth_getBlockReceipts
+			return common.NewErrEndpointClientSideException(
 				common.NewErrJsonRpcExceptionInternal(
 					int(code),
-					common.JsonRpcErrorUnsupportedException,
+					common.JsonRpcErrorInvalidArgument,
 					err.Message,
 					nil,
 					details,
 				),
 			)
 		} else if code == -32600 {
 			if dt, ok := err.Data.(map[string]interface{}); ok {
-				if msg, ok := dt["message"]; ok {
-					if strings.Contains(msg.(string), "validation errors in batch") {
-						// Intentionally return a server-side error for failed requests in a batch
-						// so they are retried in a different batch.
-						// TODO Should we split a batch instead on json-rpc client level?
-						return common.NewErrEndpointServerSideException(
+				if innerMsg, ok := dt["message"]; ok {
+					if strings.Contains(innerMsg.(string), "validation errors in batch") {
+						// Return a retryable client-side error so the caller might retry or split the batch.
+						return common.NewErrEndpointClientSideException(
 							common.NewErrJsonRpcExceptionInternal(
 								int(code),
-								common.JsonRpcErrorServerSideException,
+								common.JsonRpcErrorInvalidArgument,
 								err.Message,
 								nil,
 								details,
 							),
-							nil,
 						)
 					}
 				}
 			}
+		} else if code == -32602 || code == -32600 ||
+			strings.Contains(msg, "param is required") ||
+			strings.Contains(msg, "Invalid Request") ||
+			strings.Contains(msg, "validation errors") ||
+			strings.Contains(msg, "invalid argument") ||
+			strings.Contains(msg, "invalid params") ||
+			strings.Contains(msg, "tx of type") {
```

Comment:
This specific error is probably for tx type support, if yes then we need to retry it on network-level

Reply:
I see it's already covered above so actually let's remove it from here

Reply:
good catch. fixed